lib/std/sort.zig

```diff
@@ -89,23 +93,30 @@ pub fn heapContext(a: usize, b: usize, context: anytype) void {
     }
 }
 
-fn siftDown(a: usize, root: usize, n: usize, context: anytype) void {
-    var node = root;
-    while (true) {
-        var child = a + 2 * (node - a) + 1;
-        if (child >= n) break;
+fn siftDown(a: usize, cur: usize, b: usize, context: anytype) void {
+    var child = (math.mul(usize, cur - a, 2) catch return) + a + 1;
+    // When we don't overflow from the multiply, the above expression equals (2*cur) - (2*a) + a + 1
+    // The `+ a + 1` is safe because:
+    //  for `a > 0` then `2a >= a + 1`.
+    //  for `a = 0`, the expression equals `2*cur+1`. `2*cur` is an even number, therefore adding 1 is safe.
 
-        // choose the greater child.
-        child += @intFromBool(child + 1 < n and context.lessThan(child, child + 1));
+    // stop if we overshot the boundary
+    if (child < b) {} else return;
 
-        // stop if the invariant holds at `node`.
-        if (!context.lessThan(node, child)) break;
+    const next_child = child + 1; // `next_child` is at most `b`, therefore no overflow is possible
 
-        // swap `node` with the greater child,
-        // move one step down, and continue sifting.
-        context.swap(node, child);
-        node = child;
+    // store the greater child in `child`
+    if (next_child < b and context.lessThan(child, next_child)) {
+        child = next_child;
     }
+
+    // stop if the invariant holds at `cur`.
```

Comment:
What invariant?

Reply:
That comment was there before, but I will add a clarification that it is a "heap invariant"

Reply:
If you want to be pedantic, use `binary heap invariant`.

Reply:
I have nothing against pedantry, but I feel like at this point we are bikeshedding comments on one of the most well-understood algorithms without really making the code any clearer. The heap is clearly binary, evidenced by the fact it only deals with a maximum of two children per iteration. I also think the "math formula" is expressed clearly enough through `if (context.lessThan(child, cur))`. I don't think adding `child < cur` to the comments would help.